vendor/bat-native-ads/src/bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_events.cc

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_events.h"
+
+#include <string>
+#include <vector>
+
+#include "base/time/time.h"
+#include "bat/ads/internal/ads_client_helper.h"
+#include "bat/ads/internal/base/instance_id_util.h"
+#include "bat/ads/internal/base/logging_util.h"
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_event_info.h"
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_events_database_table.h"
+
+namespace ads {
+
+void LogTextEmbeddingHTMLEvent(const std::string embedding_formatted,
+                               const std::string hashed_key,
+                               TextEmbeddingHTMLEventCallback callback) {
+  TextEmbeddingEventInfo text_embedding_event_info;
+  text_embedding_event_info.timestamp = base::Time::Now();
+  text_embedding_event_info.version = "";
+  text_embedding_event_info.locale = "";
+  text_embedding_event_info.hashed_key = hashed_key;
+  text_embedding_event_info.embedding = embedding_formatted;
+
+  database::table::TextEmbeddingHTMLEvents database_table;
+  database_table.LogEvent(
+      text_embedding_event_info,
+      [callback](const bool success) { callback(success); });
+}
+
+void PurgeStaleTextEmbeddingHTMLEvents(
+    TextEmbeddingHTMLEventCallback callback) {
+  database::table::TextEmbeddingHTMLEvents database_table;
+  database_table.PurgeStale(
+      [callback](const bool success) { callback(success); });
+}
+
+void GetTextEmbeddingEventsFromDatabase() {
```

Comment:
from what I'm seeing in https://github.com/brave/brave-core/blob/5e0037ec19ed61ec7d2a792aa29b989e3a67b2bd/vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/pipelines/notification_ads/eligible_notification_ads_v2.cc#L60 we could either pass a callback function to `GetTextEmbeddingEventsFromDatabase` and then call it from within the `GetAll` callback. That way whoever is calling `GetTextEmbeddingEvents...` can do whatever with them in the callback function

Reply:
Added. Will push with next commit